src/web/mod.rs

```diff
@@ -278,27 +277,23 @@ fn match_version(conn: &Connection, name: &str, version: Option<&str>) -> Option
         rows.map(|row| (row.get(1), row.get(2))).collect()
     };
 
-    // first check for exact match
-    // we can't expect users to use semver in query
-    for version in &versions {
-        if version.0 == req_version {
-            return Some(MatchVersion {
-                corrected_name,
-                version: MatchSemver::Exact(version.clone()),
-            });
-        }
+    // first check for exact match, we can't expect users to use semver in query
+    if let Some((version, id)) = versions.iter().find(|(vers, _)| vers == &req_version) {
+        return Some(MatchVersion {
+            corrected_name,
+            version: MatchSemver::Exact((version.to_owned(), *id)),
+        });
     }
 
     // Now try to match with semver
     let req_sem_ver = VersionReq::parse(&req_version).ok()?;
 
     // we need to sort versions first
     let versions_sem = {
-        let mut versions_sem: Vec<(Version, i32)> = Vec::new();
+        let mut versions_sem: Vec<(Version, i32)> = Vec::with_capacity(versions.len());
 
         for version in &versions {
-            // in theory a crate must always have a semver compatible version
-            // but check result just in case
+            // in theory a crate must always have a semver compatible version, but check result just in case
```

Comment:
What's the reason for this change? It's now over 100 columns.

Reply:
There's not really a reason for it to be two, it's one continuous thought